### PR TITLE
VCPKG Compile Script for Windows

### PR DESCRIPTION
**Pull request by pvagnozzi**
choco_setup.bat: script for install choco with old terminal.
choco_setup.ps1: script for install choco with power shell.
env_setup.ps1: script for install toolchain
compile.bat: script for compile the desktop client on windows using vcpkg and cmake.

### CODE REVIEW

**Review by claucambra — COMMENTED** [review]
I am not familiar with these tools so I have added team members who are familiar with them to review

File: vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "nextcloud-desktop",
+  "version-string": "1.0.1",
+  "port-version": 1,
+  "homepage": "https://ftp.gnome.org/pub/GNOME/sources/pango/",
```
Comment (claucambra) [inline vcpkg.json:5]: This is not Nextcloud's homepage :)

**Review by claucambra — CHANGES_REQUESTED** [review]

**Review by mgallien — CHANGES_REQUESTED** [review]
thanks
I would need to do a real test as I am not familiar with VCPKG
can you move the new files into a subdirectory inside `admin/win` (for example `admin/win/vcpkg`)

File: vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "description": "Nextcloud Desktop client.",
+  "dependencies": [
+    "qt5",
+    "qtkeychain",
+    "qt5-websockets",
+    "qt5-webview",
```
Comment (mgallien) [inline vcpkg.json:11]: we use qt5-webengine so I think this one is wrong

Comment (mgallien) [reply vcpkg.json:11]: see for example the current state as reported by cmake
```
-- The following OPTIONAL packages have been found:

 * Qt5WebChannel (required version >= 5.15)
 * Qt5Positioning (required version >= 5.15)
 * Qt5WebEngineCore (required version >= 5.15)
 * Qt5PrintSupport (required version >= 5.15)
 * Qt5LinguistTools
 * IcoTool

-- The following REQUIRED packages have been found:

 * OpenSSL (required version >= 1.1)
 * ZLIB
 * Qt5Concurrent (required version >= 5.15.0), Qt5 Concurrent component.
 * Qt5WebEngineWidgets (required version >= 5.15.0), Qt5 WebEngineWidgets component.
 * Qt5WebEngine (required version >= 5.15.0), Qt5 WebEngine component.
 * Qt5Keychain
 * SQLite3 (required version >= 3.8.0)
 * Qt5WebSockets
 * Qt5Widgets
 * Qt5Svg
 * Qt5QuickControls2
 * KF5Archive
 * Qt5Core, Qt5 Core component.
 * Qt5Test
 * Qt5Xml, Qt5 Xml component.
 * Qt5Network, Qt5 Network component.
 * Qt5Qml
 * Qt5QmlModels (required version >= 5.15.10)
 * Qt5Gui (required version >= 5.15.10)
 * Qt5Quick
 * Qt5
```

File: compile.bat [hunk 3]
```diff
@@ -0,0 +1,5 @@
+@echo off
+mkdir build
+cd build
+cmake .. "-DCMAKE_TOOLCHAIN_FILE=C:/tools/vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_INSTALL_PREFIX=~/nextcloud-desktop-client -DCMAKE_BUILD_TYPE=Debug
+cmake --build . --config Debug --target install
```
Comment (mgallien) [inline compile.bat:5]: not sure you would want debug builds by default

**Review by mgallien — CHANGES_REQUESTED** [review]

**Pull request closed without merging**